test/unittest/tests/block/test_block_api_write.cpp

```diff
@@ -0,0 +1,279 @@
+
+        // The data should match the expected string.
+        const auto &str = expected_strings.at(i);
+        CHECK(memcmp(str.data(), WT_BLOCK_HEADER_BYTE(read_buf.mem), str.size()) == 0);
+        __wt_buf_free(nullptr, &read_buf);
+    }
+}
+
+void
+create_write_buffer(WT_BM *bm, std::shared_ptr<mock_session> session, std::string contents,
+  WT_ITEM *buf, size_t buf_memsize)
+{
+    // Fetch write buffer size from block manager.
+    REQUIRE(bm->write_size(bm, session->get_wt_session_impl(), &buf_memsize) == 0);
+    test_and_validate_write_size(bm, session, buf_memsize);
+
+    // Initialize the buffer with aligned size.
+    F_SET(buf, WT_ITEM_ALIGNED);
+    REQUIRE(__wt_buf_initsize(session->get_wt_session_impl(), buf, buf_memsize) == 0);
+
+    /*
+     * Copy content string into the buffer.
+     *
+     * Following the architecture guide, it seems that the block manager expects a block header. I
+     * have tried to mimic that here.
+     */
+    REQUIRE(__wt_buf_grow_worker(session->get_wt_session_impl(), buf, buf->size) == 0);
+    memcpy(WT_BLOCK_HEADER_BYTE(buf->mem), contents.c_str(), contents.length());
+}
+
+TEST_CASE("Block manager: file operation read, write and write_size functions", "[block_api]")
+{
+    // Build Mock session, this will automatically create a mock connection.
+    std::shared_ptr<mock_session> session = mock_session::build_test_mock_session();
+    config_parser cp({{"allocation_size", ALLOCATION_SIZE}, {"block_allocation", BLOCK_ALLOCATION},
+      {"os_cache_max", OS_CACHE_MAX}, {"os_cache_dirty_max", OS_CACHE_DIRTY_MAX},
+      {"access_pattern_hint", ACCESS_PATTERN}});
+
+    REQUIRE(
+      (session->get_mock_connection()->setup_block_manager(session->get_wt_session_impl())) == 0);
+    session->setup_block_manager_file_operations();
+
+    WT_BM bm;
+    WT_CLEAR(bm);
+    __ut_bm_method_set(&bm);
+
+    auto path = std::filesystem::current_path();
+    path += "/test.wt";
+    REQUIRE(__wt_block_manager_create(
+              session->get_wt_session_impl(), path.c_str(), DEFAULT_BLOCK_SIZE) == 0);
+    REQUIRE(__wt_block_open(session->get_wt_session_impl(), path.c_str(), WT_TIERED_OBJECTID_NONE,
+              cp.get_config_array(), false, false, false, DEFAULT_BLOCK_SIZE, &bm.block) == 0);
+    REQUIRE(__wti_block_ckpt_init(session->get_wt_session_impl(), &bm.block->live, nullptr) == 0);
+
+    SECTION("Test write_size api")
+    {
+        test_and_validate_write_size(&bm, session, 0);
+        test_and_validate_write_size(&bm, session, 800);
+        test_and_validate_write_size(&bm, session, 1234);
+        test_and_validate_write_size(&bm, session, 5000);
+        test_and_validate_write_size(&bm, session, 5120);
+        test_and_validate_write_size(&bm, session, 9999);
+
+        size_t init_size = UINT32_MAX - 1000;
+        REQUIRE(bm.write_size(&bm, session->get_wt_session_impl(), &init_size) == EINVAL);
+    }
+
+    SECTION("Test generic write api")
```

Comment:
Read properly this makes it sound like there are multiple versions of the write api, one which does "generic" writes and further down one which does "complex" writes. A better test case description would be: "Test writing a single string using the block manager".

Then remove all references to "generic".